### ISSUE TITLE
IndieWeb Utils v0.3.0

### ISSUE THREAD

**Issue by capjamesg** [capjamesg/indieweb-utils#56]
The PRs currently in progress, as well as those worked on over the last few months after the release of v0.2.0, have added many new changes to the library. We have improved documentation, worked on an I/O refactor, fixed bugs, and added a few changes to the library. I was wondering if this is the time to prepare to launch v0.3.0.

To make this launch happen, we should first finish and then close all PRs in progress. Then, I can prepare a changelog update with all of the revisions we have made to the library since the last version.

@jamesvandyne Do you think we're ready to upgrade the version of IndieWeb Utils?

**Comment by capjamesg** [comment]
Here is a stub changelog summarising the changes so far (that are not in PRs):

```
## [0.3.0] - TBD

### Added

#### Development

- Provide docstrings for all functions in the library that did not have a docstring.
- Fix docstring rendering issues with library documentation so that all docstrings show up on [Read the Docs](https://indieweb-utils.readthedocs.io/en/latest/).
- Add `:raises:` statements to docstrings to document existing
- Add code examples to docstrings and remove redundant examples from RS documentation.

#### Functions

- `discover_h_feed()` function to discover the representative h-feed on a page.
- `get_valid_relmeauth_links()` function to find both one-way and bi-directional rel=me links on a web page.
- `get_representative_h_card()` function to get the [representative h-card](https://microformats.org/wiki/representative-h-card-parsing) associated with a web page.

#### Tests

- Added test cases for:
    - `discover_h_feed()`
    - `get_representative_h_card()`
    - `get_valid_relmeauth_links()`

### Changed

- Support importing IndieAuth functions directly from `indieweb_utils` without having to use `indieweb_utils.indieauth.`.
- Simplify `get_h_app_item()` logic.
- Raise `HAppNotFound` exception when `get_h_app_item()` cannot identify a h-app microformat.
- `discover_webmention_endpoint()` can now raise LocalhostEndpointFound, TargetNotProvided, UnacceptableIPAddress, and WebmentionEndpointNotFound exceptions when there is an issue validating a webmention.
- `send_webmention()` can now raise MissingSourceError, MissingTargetError, UnsupportedProtocolError, TargetIsNotApprovedDomain, GenericWebmentionError, and CouldNotConnectToWebmentionEndpoint if there was an issue sending a webmention.
- `send_webmention()` now returns the HTTP status code and headers of a successful webmention.
```

**Comment by capjamesg** [comment]
Here are a few tasks we need to complete when we're ready:

1. Change version numbers.
2. Update changelog with all the new features added in recent PRs.
3. Reorganise the documentation a bit more since recent PRs include additions to the newly organised documentation sections.

**Comment by capjamesg** [comment]
We have released v0.3.0 and v0.3.1. The v0.3.0 contains all of the changes we have worked on, and .1 contains a minor fix to ensure PyPi could interpret our README file.

Release notes:

https://github.com/capjamesg/indieweb-utils/releases/tag/v0.3.0
https://github.com/capjamesg/indieweb-utils/releases/tag/v0.3.1

I am going to close this issue since we released the new version.
